kernel/kernel-api/src/main/java/io/delta/kernel/internal/actions/Format.java

```diff
@@ -79,4 +79,20 @@ public Row toRow() {
 
         return new GenericRow(Format.FULL_SCHEMA, formatMap);
     }
+
+    @Override
+    public String toString() {
+        StringBuilder sb = new StringBuilder();
+        sb.append("Format(");
+        sb.append(provider).append(",");
+        sb.append("Map(");
+        for (Map.Entry<String, String> entry : options.entrySet()) {
+            sb.append(entry.getKey()).append(" -> ").append(entry.getValue()).append(", ");
+        }
+        if (sb.substring(sb.length() - 2).equals(", ")) {
+            sb.setLength(sb.length() - 2);
+        }
+        sb.append("))");
+        return sb.toString();
+    }
```

Comment:
```suggestion
    @Override
    public String toString() {
        return "Format{" +
                "provider='" + provider + '\'' +
                ", options=" + options +
                '}';
    }
```

Reply:
Same comment for the `Metadata.toString`. You can actually automatically generate these using the Intellij.